pkg/operator/ceph/object/notification/s3ext.go

```diff
@@ -66,21 +66,24 @@ func (s *DeleteBucketNotificationRequestInput) Validate() error {
 
 const opDeleteBucketNotification = "DeleteBucketNotification"
 
-func DeleteBucketNotificationRequest(c *s3.S3, input *DeleteBucketNotificationRequestInput) *request.Request {
+func DeleteBucketNotificationRequest(c *s3.S3, input *DeleteBucketNotificationRequestInput, notificationId string) *request.Request {
 	op := &request.Operation{
 		Name:       opDeleteBucketNotification,
 		HTTPMethod: http.MethodDelete,
 		HTTPPath:   "/{Bucket}?notification",
 	}
 
+	if len(notificationId) > 0 {
+		op.HTTPPath = "/{Bucket}?notification=" + notificationId
+	}
 	if input == nil {
 		input = &DeleteBucketNotificationRequestInput{}
 	}
 
 	return c.NewRequest(op, input, nil)
 }
 
-func DeleteBucketNotification(c *s3.S3, input *DeleteBucketNotificationRequestInput) error {
-	req := DeleteBucketNotificationRequest(c, input)
+func DeleteBucketNotification(c *s3.S3, input *DeleteBucketNotificationRequestInput, notificationId string) error {
+	req := DeleteBucketNotificationRequest(c, input, notificationId)
 	return req.Send()
 }
```

Comment:
Tell me more about these notification IDs. Are notification IDs always unique, or do they change/shift when other notifications are deleted?

Reply:
@yuvalif ^^

Reply:
they are unique per bucket. but not over time - the same value could be reused after it has been deleted

Reply:
With this information, the extra possibility this feature has for race conditions is making me increasingly nervous.

Consider that bucket notifications can have two simultaneous reconciles ongoing at the same time. One can be a result of changes to OBCs and the other from a CephObjectBucketNotification resource. I realize that race conditions in the wild are going to be hard to reach, but it seems possible that the OBC reconcile process and Notification reconcile process could both run at the same time and interact in ways we don't expect. What if process A creates a an RGW notification that should exist, but a slower process B responds to an older request to delete the RGW notification right after, resulting in process A thinking that it has configured the notification when in fact it was later deleted without its knowing?

I would suggest that we should have a mutex lock to prevent both from running simultaneously so we can avoid getting confused bug reports from users or testers. It might not be pretty, but it will at least prevent race conditions that result in states due to interactions we don't foresee.

Reply:
Current PR does not have this change, I will add it next update

Reply:
not sure how this is currently possible. we, do not support CephObjectBucketNotification deletions
once we add that functionality we will have to take care of possible race conditions

Reply:
I realize I was not clear that I'm not asking for a race condition solution (like mutex) in this PR. Sorry about the confusion. I think it is best left for its own PR. Thanks both of you. I'll take another look later today.